Convert Cesium plugin to use goog.module

schmidtk opened the pull request:
The entire Cesium plugin has been converted to use `goog.module`. I cleaned up somethings we were doing in the plugin that resulted in compiler errors using ES classes.

- Some Cesium classes like `Cesium.TilingScheme` are really abstract classes and don't provide any functional code. The Cesium class/object is not available until the library is loaded, so we had code to defer creating extending classes until that happens. This was a little strange with modules/classes, so I changed the Cesium externs to make the original constructor an interface instead. This is much cleaner, and has no side effects given Cesium's implementations were all empty anyway.
- Some Cesium mixins also needed to load async after the library loaded, so I cleaned up those files to a more consistent and module-friendly pattern.
- We frequently used `Object.defineProperties` to add `get/set` functions to Cesium plugin classes. ES classes can use `get/set` directly, so I refactored all of those so the resulting code is much cleaner.
- We had some janky code to replace the OLCS camera class with our own, because we didn't want to use their view listener. We were using all of the other code, so I dropped the class replace and instead replaced `setView_` (the view change handler) on the OLCS class with an empty function. This has the same end result, but in a less janky manner.

jsalankey commented on src/plugin/cesium/cesium.js:
```diff
@@ -540,68 +518,48 @@ plugin.cesium.hasWorldTerrain = function() {
   const renderer = map.getWebGLRenderer();
   if (renderer) {
     const supported = renderer.getSupportedTerrainProviders();
     const worldTerrain = supported.find((p) => p.type === os.map.terrain.TerrainType.ION);
```
Still using a global reference for `TerrainType` here.

schmidtk replied:
I started converting some globals, but we should write a transform to do that for us in an intelligent manner using the `deps.js` file. Otherwise it's going to add a large amount of tedious extra work across all of OpenSphere's sources.

jsalankey commented on src/plugin/cesium/cesiumcamera.js:
```diff
@@ -377,545 +426,528 @@ plugin.cesium.Camera.prototype.initExtentComputation_ = function() {
+
+      var target;
+      if (options.center) {
+        var center = proj.transform(options.center, osMap.PROJECTION, os.proj.EPSG4326);
```
Another global reference for `os.proj`. Given how long this PR is, there are probably a few of these left. Not sure if we want to bother picking through it for them or just dealing with them later.

schmidtk replied:
Yeah we can write a transform to do these later.

jsalankey commented on src/plugin/cesium/cesium.js:
```diff
@@ -145,54 +145,51 @@ plugin.cesium.addTrustedServer = function(url) {
-    os.ui.window.launchConfirmText(/** @type {!osx.window.ConfirmTextOptions} */ ({
+const promptForAccessToken = function() {
+  return new Promise(function(resolve, reject) {
+    ui.window.launchConfirmText(/** @type {!osx.window.ConfirmTextOptions} */ ({
```
Another `ui.window` case.

schmidtk replied:
Thanks, fixed.

schmidtk commented on src/plugin/cesium/cesiummenu.js:
```diff
@@ -51,6 +50,12 @@ plugin.cesium.menu.launchAddIonAsset = function() {
-  var template = '<importionasset></importionasset>';
-  os.ui.window.create(windowOptions, template);
+  var template = `<${importIonAssetTag}></${importIonAssetTag}>`;
+  ui.window.create(windowOptions, template);
```
These should be fixed though, because they're referencing the wrong module.

schmidtk replied:
Fixed.

schmidtk commented:
There were some functions in `src/os/ui/icons.js` that were on the `os.ui` namespace, which caused similar problems to the noted `os.ui.window` references. To resolve this and hopefully better prep for converting that file, I moved all functions to `os.ui.icons`, provided that namespace, and added deprecated references to the old function paths. I fixed all references in OpenSphere, and external projects will see a deprecation warning if they're using the old functions.
